### PR TITLE
Desktop backgrounding support

### PR DESCRIPTION
**Pull request by gnunicorn**
built on top of #2028 , this adds:
1. support to keep the desktop app running upon "closing" the window
2. a tray icon to resurrect the window from the backgrounded process

Due to the nature of he different platforms, this acts slightly different on each one of them:

On MacOS, we currently only support the toggle:

https://github.com/user-attachments/assets/250d3b75-930e-493b-869d-7670d865e152

On Windows, we have a colored icon and toggle as well as menu support:

https://github.com/user-attachments/assets/513f28cb-6550-4da3-8486-5ff50be5d9a1

On Linux, toggle doesn't exist on the icon itself, but only via the menu:

https://github.com/user-attachments/assets/786448e7-c882-4a94-9835-f45d04cadb58

### CODE REVIEW

File: app/windows/runner/main.cpp [hunk 1]
```diff
@@ -7,6 +7,13 @@
 
 int APIENTRY wWinMain(_In_ HINSTANCE instance, _In_opt_ HINSTANCE prev,
                       _In_ wchar_t *command_line, _In_ int show_command) {
+
+  HWND hwnd = ::FindWindow(L"FLUTTER_RUNNER_WIN32_WINDOW", L"Acter");
+  if (hwnd != NULL) {
+    ::ShowWindow(hwnd, SW_NORMAL);
+    ::SetForegroundWindow(hwnd);
+    return EXIT_FAILURE;
+  }
```
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/windows/runner/main.cpp:16]: yes, for the tray support it only makes sense if there isn't multiple instances.

[the remark this answers is not in the document]
Comment (bitfriend) [reply app/windows/runner/main.cpp:16]: It is reasonable to rename process to `Acter` too.
image

**Comment by bitfriend** [comment]
In windows, I couldn't open context menu on system tray.
And system tray shows tooltip, but doesn't show acter icon.

File: app/windows/runner/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (gnunicorn) [reply app/windows/runner/main.cpp:16]: @bitfriend not the focus of this PR but if you tell me _how_ I can add it...

**Review by bitfriend — APPROVED** [review]
Great solution.
Just confirmed it works fine on windows.

File: app/windows/runner/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bitfriend) [reply app/windows/runner/main.cpp:16]: Fixed by https://github.com/acterglobal/a3/pull/2070

**Pull request merged**